openedx/adg/lms/applications/admin.py

```diff
@@ -122,14 +127,59 @@ class WorkExperienceAdmin(admin.ModelAdmin):
     search_fields = ('name_of_organization', 'job_position_title',)
 
 
-class PrerequisiteCourseAdmin(admin.TabularInline):
-    model = PrerequisiteCourse
+class MultilingualCourseInlineFormset(BaseInlineFormSet):
+    """
+    Inline formset for multilingual courses
+    """
+
+    def clean(self):
+        super(MultilingualCourseInlineFormset, self).clean()
+
+        course_group_lang_count = defaultdict(lambda: 0)
+        store = modulestore()
+
+        for form in self.forms:
+            if not form.cleaned_data:
+                continue
+
+            course_info = store.get_course(form.instance.course.id)
+            if not course_info.language:
+                raise ValidationError(_('Please add language of the course from studio.'))
+
+            course_group_lang_count[course_info.language] += 1
+            if course_group_lang_count[course_info.language] > 1:
+                raise ValidationError(
+                    _(
+                        'You cannot add 2 or more courses with same language.'
+                        ' Add courses of different language or change language of a course from studio!'
+                    )
+                )
+
+
+class MultilingualCourseAdmin(admin.TabularInline):
+    """
+    Inline admin for Multilingual Courses
+    """
+
+    model = MultilingualCourse
+    formset = MultilingualCourseInlineFormset
+    NO_OF_INLINE_FORM_EMPTY_GROUP = 2
+    NO_OF_INLINE_FORM_NON_EMPTY_GROUP = 0
 
+    def get_extra(self, request, obj=None, **kwargs):
+        """
+        Customized number of inline forms
+        """
+        if obj and obj.multilingual_courses.exists():
+            return self.NO_OF_INLINE_FORM_NON_EMPTY_GROUP
+        else:
+            return self.NO_OF_INLINE_FORM_EMPTY_GROUP
 
-@admin.register(PrerequisiteCourseGroup)
-class PrerequisiteCourseGroupAdmin(admin.ModelAdmin):
-    inlines = [PrerequisiteCourseAdmin, ]
-    list_display = ('name', 'prereq_course_count', 'open_prereq_courses_count',)
+
+@admin.register(MultilingualCourseGroup)
+class MultilingualCourseGroupAdmin(admin.ModelAdmin):
+    inlines = (MultilingualCourseAdmin,)
+    list_display = ('name', 'is_prerequisite', 'multilingual_course_count', 'open_multilingual_courses_count')
```

Comment:
Kindly make use of trailing comma consistent; either remove it from everywhere or add it everywhere.

Reply:
Isn't it consistent? At line 181, the comma is required to define a tuple as it has a single element but for line 182 we don't need comma. For tuples with single element comma is required to differentiate them but for multiple elements it isn't required.

Reply:
Also the trailing comma is helpful if a tuple, list or a dict spans over multiple lines.

Reply:
We've been using a trailing comma in some of our code. I thought maybe we could make a convention either way.

Reply:
You are right, We can have some convention on this as well.